ts/Series/Heatmap/HeatmapSeries.ts

```diff
@@ -426,18 +435,67 @@ class HeatmapSeries extends ScatterSeries {
      * @private
      */
     public drawPoints(): void {
-
-        // In styled mode, use CSS, otherwise the fill used in the style sheet
-        // will take precedence over the fill attribute.
-        const seriesMarkerOptions = this.options.marker || {};
-
-        if (seriesMarkerOptions.enabled || this._hasPointMarkers) {
-            Series.prototype.drawPoints.call(this);
-            this.points.forEach((point): void => {
+        const heatmap = this,
+            // In styled mode, use CSS, otherwise the fill used in the style
+            // sheet will take precedence over the fill attribute.
+            seriesMarkerOptions = heatmap.options.marker || {};
+
+        if (heatmap.options.interpolation) {
+            const chart = heatmap.chart,
+                ctx = heatmap.getContext(),
+                canvas = heatmap.canvas,
+                colorAxis = chart.colorAxis && chart.colorAxis[0];
+
+            if (canvas && ctx && colorAxis) {
+                const pixels =
+                    ctx.createImageData(canvas.width, canvas.height),
+                    image = heatmap.image;
+
+                let pixelPos = 0,
+                    color,
+                    rgb;
+
+                heatmap.data.forEach((p: HeatmapPoint): void => {
```

Comment:
This `data.forEach` must be replaced by logic that finds the actual position of the data points on the heatmap. What this code currently does is to just fill in pixels in the canvas successively. In the current demo, when toggling the `Interpolation`, we can see that it is not correct.

That's because the data is not necessarily sorted, so when the points are applied to pixels, each point's pixel position must be determined individually. Use the `x` and `y` values and the `colsize` and `rowsize` options as well axis extremes.